src/ajax/core.cljs

```diff
@@ -345,10 +347,37 @@
     (str/upper-case (name method))
     method))
 
-(defn process-inputs [{:keys [uri method format params headers]}
+(defn add-default-interceptor
+  "I.E. (add-default-interceptor {:request (fn [[uri body headers]]
+                                               (do-something-to-request uri body headers))
+                                  :response (fn [resp]
+                                                (do-something-to-response resp))})"
+  [interceptor]
+  (swap! default-interceptors conj interceptor))
+
+(defn apply-interceptors
+  "Apply each interceptor in the `interceptors` list to the request / response."
+  [stage interceptors input]
+  (loop [remaining (reverse interceptors)
+         output input]
+    (if (empty? remaining)
+      output
+      (let [interceptor (or (->> remaining
+                                 (filter stage)
+                                 (map stage)
+                                 (first))
+                            identity)]
+        (recur (rest remaining)
+               (interceptor output))))))
+
+(defn process-inputs [{:keys [uri method format params headers interceptors]}
                       {:keys [content-type]}]
   (let [headers (merge {"Accept" content-type}
-                       (or headers {}))]
+                       (or headers {}))
+        [uri method format params headers] (apply-interceptors :request
```

Comment:
So like this instead?

~~~clojure
(defn ajax-request
  [{:keys [method api interceptors] :as opts}]
  (let [response-format (get-response-format opts)
        method (normalize-method method)
        [uri body headers] (process-inputs (apply-interceptors :request
                                                               (or interecptors @default-interceptors)
                                                               opts)
                                           response-format)
        handler (base-handler response-format opts)
        api (or api (new goog.net.XhrIo))]
    (-js-ajax-request api uri method body
                      headers handler opts)))
~~~